vscode-car-plugin/src/main/java/org/wso2/maven/CARMojo.java

```diff
@@ -0,0 +1,230 @@
+     */
+    private void appendLogs() {
+        getLog().info("------------------------------------------------------------------------");
+        getLog().info("Building Synapse Config Project");
+        getLog().info("------------------------------------------------------------------------");
+    }
+
+    private File getArchiveFile(String fileExtension) {
+        File archiveFile;
+        if (finalName != null && !finalName.trim().equals("")) {
+            archiveFile = new File(archiveLocation, finalName + fileExtension);
+            return archiveFile;
+        }
+        String archiveFilename = project.getArtifactId() + "_" + project.getVersion() +
+                (classifier != null ? "-" + classifier : "") + fileExtension;
+        archiveFile = new File(archiveLocation, archiveFilename);
+
+        return archiveFile;
+    }
+
+    private void zipFolder(String srcFolder, String destZipFile) {
+        try {
+            FileOutputStream fileWriter = new FileOutputStream(destZipFile);
+            ZipOutputStream zip = new ZipOutputStream(fileWriter);
+
+            addFolderContentsToZip(srcFolder, zip);
+
+            zip.flush();
+            zip.close();
+        } catch (Exception ex) {
+            getLog().error("Error occurred while creating CAR file. " + ex);
+        }
+    }
+
+    private void addFolderContentsToZip(String srcFolder, ZipOutputStream zip) {
+        File folder = new File(srcFolder);
+        String[] fileList = folder.list();
+        try {
+            if (fileList != null) {
+                int i = 0;
+                while (true) {
+                    if (fileList.length == i) break;
+                    if (new File(folder, fileList[i]).isDirectory()) {
+                        zip.putNextEntry(new ZipEntry(fileList[i] + "/"));
+                        zip.closeEntry();
+                    }
+                    addToZip("", srcFolder + "/" + fileList[i], zip);
+                    i++;
+                }
+            }
+        } catch (Exception ex) {
+            getLog().error("Error occurred while creating CAR file. " + ex);
+        }
+    }
+
+    private void addToZip(String path, String srcFile, ZipOutputStream zip) {
+        File folder = new File(srcFile);
+        if (folder.isDirectory()) {
+            addFolderToZip(path, srcFile, zip);
+        } else {
+            // Transfer bytes from in to out
+            byte[] buf = new byte[1024];
+            int len;
+            try {
+                FileInputStream in = new FileInputStream(srcFile);
+                if (path.trim().equals("")) {
+                    zip.putNextEntry(new ZipEntry(folder.getName()));
+                } else {
+                    zip.putNextEntry(new ZipEntry(path + "/" + folder.getName()));
+                }
+                while ((len = in.read(buf)) > 0) {
+                    zip.write(buf, 0, len);
+                }
+                in.close();
```

Comment:
Better to move closing resources to finally block

Reply:
Corrected according to [try-with-resources Statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) in java